cfgloader/cfgloader.go

```diff
@@ -105,6 +112,10 @@ func Load(ctx context.Context, cfg any, opt ...Option) error {
 		return fmt.Errorf("failed to load config: %w", err)
 	}
 
+	if d, ok := cfg.(Defaultable); ok {
```

Comment:
I think all of your new test cases test the positive path here (cfg implements Defaultable), right? If you want to be thorough you could test the negative path also.

Reply:
Good point. Let me add.

Reply:
Separated validatable, defaultable tests